crates/python/src/lib.rs

```diff
@@ -68,3 +69,9 @@ fn reset_logging() {
         }
     }
 }
+
+#[pyfunction]
+#[pyo3(name = "_gather_diagnostics")]
+fn gather_diagnostics(py: Python<'_>) -> PyResult<String> {
+    py_sync::py_sync!(py, async { Ok(qcs::diagnostics::get_report().await) })
+}
```

Comment:
Decided to only export a synchronous version of this function to keep the API simpler.